utils/rpms/daos.spec

```diff
@@ -89,6 +89,7 @@ Requires: libpmem1, libpmemobj1
 %endif
 Requires: fuse >= 3.4.2
 Requires: protobuf-c
+Requires: fio < 3.4
```

Comment:
Does this really need to be `< 3.4`?  EL7 provides `fio` in it's on repos:

```
fio-0:3.1-1.el7.x86_64
fio-0:3.3-5.el7.x86_64
fio-0:3.7-1.el7.x86_64
```

as does Leap 15:

```
Information for package fio:
----------------------------
Repository     : osl151-oss                                      
Name           : fio                                             
Version        : 3.4-lp151.4.6                                   
```

Reply:
do we still need to rely on fio?

Reply:
Commit message for adding it back in says: "reinstate fio in build , as it's used in CI" so it seems it might be used in testing.

In any case, putting the requirement in the main package is wrong.  At most, it should be in the _tests_ subpackage.

Reply:
I don't really want to change the version as part of this patch, it's not in the scope of this PR. I'm just reverting the removal of fio in this PR because the CI tests failed when I removed it (because FioSmall tests rely on the package being part of the build).

Reply:
we should change the version and Move where the packages referenced in a separate PR, this PR should be focused on the SPDK upgrade.

Reply:
addressed in https://github.com/daos-stack/daos/pull/2339